docs/usage.md

```diff
@@ -111,20 +109,60 @@ Example:
 qsimh_base just computes and just prints the first eight amplitudes. The hybrid
 Schrödinger-Feynman method is used. The lattice is split into two parts.
 A two level checkpointing scheme is used to improve performance. Say, there
-are `k` gates on the cut. We split those into three parts: `p+r+s=k`, where
+are `N` gates on the cut. We split those into three parts: `p+r+s=N`, where
 `p` is the number of "prefix" gates, `r` is the number of "root" gates and
 `s` is the number of "suffix" gates. The first checkpoint is executed after
 applying all the gates up to and including the prefix gates and the second
 checkpoint is executed after applying all the gates up to and including the
 root gates. The full summation over all the paths for the root and suffix gates
-is performed. The path for the prefix gates is specified by `prefix`. It is
-just a value of bit-shifted path indices in the order of occurrence of prefix
-gates in the circuit file.
+is performed.
 
-Example:
+The path for the prefix gates is specified by `prefix`. It is just a value of
+bit-shifted path indices in the order of occurrence of prefix gates in the
+circuit file. This is primarily used for distributed execution - see the
+`Distributed execution` section below for more details.
+
+Example (running on one machine):
+```
+./qsimh_base.x -c ../circuits/circuit_q30 -d 16 \
+               -k 0,1,2,6,7,8,12,13,14,18,19,20,24,25,26 \
+               -t 8 -w 0 -p 0 -r 5 -v 1
 ```
-./qsimh_base.x -c ../circuits/circuit_q30 -d 16 -k 0,1,2,6,7,8,12,13,14,18,19,20,24,25,26 -t 8 -w 0 -p 0 -r 5 -v 1
+
+### Choosing flag values for qsimh
+
+**-k** defines how the lattice will be split up. In the examples above, the
+lattice has the structure below (cuts are denoted by the `|` symbol):
+
 ```
+ 0    1    2 |  3    4    5
+
+ 6    7    8 |  9   10   11
+
+12   13   14 | 15   16   17
+
+18   19   20 | 21   22   23
+
+24   25   26 | 27   28   29
+```
+
+Deciding which cuts are optimal for a given circuit is computationally hard.
+However, splitting the grid into roughly equal parts with the fewest cuts
+possible (as is done for the lattice above) produces a circuit that performs
+reasonably well in most cases.
+
+**-p** and **-r** specify the number of "prefix" and "root" gates respectively,
+and determine how much of the circuit must be repeated for each execution.
+Values for these flags should be chosen with this in mind:
+
+- Operations up to and including the "prefix" gates will only be run once, but
```

Comment:
These lines might be a bit confusing.

Reply:
Removed a line - the list below explains it better. Let me know if this didn't fix your concern.

Reply:
Perhaps I would add something like the following:

The runtime can be controlled by **-p**. There is no summation over the "prefix" gates and the unique "prefix" path is specified by **-w**, see the `Distributed execution` section below for details on this. **-r** specifies implicitly the number of the "suffix" gates: the total number of gates on the cut minus the values specified by **-p** and **-r**. Typically for performance, the "suffix" gates should be the gates on the cut with maximum "time".

Reply:
Done.